src/Compatibility/Microsoft.DotNet.ApiCompatibility/Abstractions/MetadataInformation.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+
+namespace Microsoft.DotNet.ApiCompatibility.Abstractions
+{
+    /// <summary>
+    /// Struct containing the assembly's relevant information, used to distinguish different tuple comparisons
+    /// and different list of <see cref="CompatDifference"/>.
+    /// </summary>
+    public readonly struct MetadataInformation : IEquatable<MetadataInformation>
+    {
+        public readonly string AssemblyName;
+        public readonly string TargetFramework;
+        public readonly string AssemblyType;
+
+        public MetadataInformation(string name, string targetFramework, string assemblyType)
+        {
+            AssemblyName = name ?? string.Empty;
+            TargetFramework = targetFramework ?? string.Empty;
+            AssemblyType = assemblyType ?? string.Empty;
```

Comment:
should we make this an enum ?

Reply:
How would that work? I was expecting this to have, `lib`, or `runtimes/win-x64`, etc. We would have to list all the possible runtimes?

Reply:
ah okay, i thought we are just storing the assembly type like, ref, lib or runtime

Reply:
No, I think it should be more specific to the runtime the assembly belongs to if it is a runtime assembly.